tools/tgs4_scripts/PreCompile.sh

```diff
@@ -90,6 +92,17 @@ make
 mv libbyond-extools.so "$1/libbyond-extools.so"
 cd ../../..
 
+# install or update youtube-dl when not present, or if it is present with pip3,
+# which we assume was used to install it
+if ! [ -x "$has_youtubedl" ]; then
+	echo "Installing youtube-dl with pip3..."
+	apt-get install -y python3 python3-pip
```

Comment:
I don't really know anything about the way our servers are setup now, but this should just install locally, right? It won't cause problems because you're not using sudo?

Reply:
good catch! i have updated the script to handle sudo just like it does for the other dependency installations.

Reply:
i notice we actually have a mix of sudo and non-sudo, should maybe review that